### PR TITLE
Implement Pause Restic

### LINKED ISSUE #279: Consider a simple 'enabled' switch for Restic CRD
I know there are [a couple ways to disable backups](https://appscode.com/products/stash/0.6.2/concepts/crds/restic/), but neither are great IMHO. I'd rather not 'lose' the CRD by deleting it, nor 'damage' the CRD (change selector or schedule) just to disable it temporarily.

Could you consider a simple `spec.enabled: true/false` flag that left the CRD intact but disabled. I guess the only way to communicate that to the sidecar is to patch it back out. But that would be fine.

There would then be an easy way to bulk disable back-ups for a period, then re-enable them later, but patching 'spec.enabled: false' into multiple CRDs.

> Disable Backup
To stop taking backup, you can do 2 things:
> 1. Delete the Restic CRD. Stash operator will remove the sidecar container from all matching workloads.
> 2. Change the labels of a workload. Stash operator will remove sidecar container from that workload. This way you can selectively stop backup of a Deployment, ReplicaSet, etc.

### PR DESCRIPTION
**Pull request by Emruz-Hossain**
Fixes #279

### CODE REVIEW

File: test/e2e/deployment_test.go [hunk 1]
```diff
@@ -741,4 +743,84 @@ var _ = Describe("Deployment", func() {
...
+				By("Waiting for backup to complete")
+				f.EventuallyRestic(restic.ObjectMeta).Should(WithTransform(func(r *api.Restic) int64 {
+					return r.Status.BackupCount
+				}, BeNumerically(">=", 1)))
+
+				By("Waiting for backup event")
+				f.EventualEvent(restic.ObjectMeta).Should(WithTransform(f.CountSuccessfulBackups, BeNumerically(">=", 1)))
+
+				By(`Patching Restic with "paused: true"`)
+				err = f.PatchRestic(&restic, func(in *api.Restic) *api.Restic {
```
Comment (diptadas) [inline test/e2e/deployment_test.go]: assign newly patched restic

File: test/e2e/deployment_test.go [hunk 2]
```diff
@@ -28,6 +29,7 @@ var _ = Describe("Deployment", func() {
 		cred       core.Secret
 		deployment apps.Deployment
 		recovery   api.Recovery
+		previousBackupCount int64
```
Comment (diptadas) [inline test/e2e/deployment_test.go]: Don't need to define it globally.

File: pkg/docker/checks.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package docker
 
 const (
-	ACRegistry   = "appscode"
+	ACRegistry   = "emruzhossain"
```
Comment (diptadas) [inline pkg/docker/checks.go]: You can set it [here](https://github.com/appscode/stash/blob/master/test/e2e/e2e_suite_test.go#L72) for running e2e tests.

File: pkg/backup/backup.go [hunk 4]
```diff
@@ -189,6 +189,9 @@ func (c *Controller) setup() (*api.Restic, error) {
 }
 
 func (c *Controller) runResticBackup(resource *api.Restic) (err error) {
+	if resource.Spec.Paused == true {
+		return nil
```
Comment (tamalsaha) [inline pkg/backup/backup.go]: log.InfoLn("skipped logging since restic is paused.")

File: pkg/controller/daemonsets.go [hunk 5]
```diff
@@ -59,7 +59,9 @@ func (c *StashController) runDaemonSetInjector(key string) error {
 		}
 
 		if newRestic != nil && !util.ResticEqual(oldRestic, newRestic) {
-			return c.EnsureDaemonSetSidecar(ds, oldRestic, newRestic)
+			if newRestic.Spec.Paused == false {
```
Comment (tamalsaha) [inline pkg/controller/daemonsets.go]: Use `!newRestic.Spec.Paused`

File: pkg/backup/backup.go (hunk 4 above)
Comment (Emruz-Hossain) [reply pkg/backup/backup.go]: added.

File: pkg/controller/daemonsets.go (hunk 5 above)
Comment (Emruz-Hossain) [reply pkg/controller/daemonsets.go]: updated

**Pull request merged**